tests/test_pdffonts.py

```diff
@@ -0,0 +1,109 @@
+import sys
+import unittest
+from tools.pdffonts import commandlineargumenthandler
+
+
+class TestPdffonts(unittest.TestCase):
+    # Test calling commandlineArgumentHandler with no args
+    def test_commandlineargumenthandlernoargs(self):
+        def usage():
+            return 100
+
+        self.assertEqual(commandlineargumenthandler([], usage=usage), 100)
+        return
+
+    # Test correct behaviour of commandlineArgumentHandler given simple1.pdf
+    def test_commandlineargumenthandler_simple1(self):
+        def usage():
+            return 100
+        expectedFontObject = {
+            "Helvetica": "Type 1"
+        }
+        sys.stdout = open("/dev/stdout", "w")
+        self.assertEqual(
+            commandlineargumenthandler(["null", "samples/simple1.pdf"], usage=usage),
+            expectedFontObject,
+        )
+        return
+
+    # Test correct behaviour of commandlineArgumentHandler given simple2.pdf
+    def test_commandlineargumenthandler_simple2(self):
+        def usage():
+            return 100
+        expectedFontObject = {
+        }
+        sys.stdout = open("/dev/stdout", "w")
+        self.assertEqual(
+            commandlineargumenthandler(["null", "samples/simple2.pdf"], usage=usage),
+            expectedFontObject,
+        )
+        return
+
+    # Test correct behaviour of commandlineArgumentHandler given simple3.pdf
+    def test_commandlineargumenthandler_simple3(self):
+        def usage():
+            return 100
+        expectedFontObject = {
+            "Helvetica": "Type 1",
+            "unknown": "Type CID"
+        }
+        sys.stdout = open("/dev/stdout", "w")
+        self.assertEqual(
+            commandlineargumenthandler(["null", "samples/simple3.pdf"], usage=usage),
+            expectedFontObject,
+        )
+        return
+
+    # Test correct behaviour of commandlineArgumentHandler given simple3.pdf with 90 degree rotation
+    def test_commandlineargumenthandler_simple3withrotation(self):
+        def usage():
+            return 100
+        expectedFontObject = {
+            "Helvetica": "Type 1",
+            "unknown": "Type CID"
+        }
+        sys.stdout = open("/dev/stdout", "w")
+        self.assertEqual(
+            commandlineargumenthandler(["null", "-R", "90", "samples/simple3.pdf"], usage=usage),
+            expectedFontObject,
+        )
+        return
+
+    # Test correct behaviour of commandlineArgumentHandler given an encrypted pdf
+    def test_commandlineargumenthandler_encrypted(self):
```

Comment:
For some reason, this test fails on my machine. When running `make test`, I get this output:
```
/home/axel/Documents/uni/maint/repo/tests/test_pdffonts.py:100: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/stdout' mode='w' encoding='UTF-8'>
  sys.stdout = open("/dev/stdout", "w")
ResourceWarning: Enable tracemalloc to get the object allocation traceback
name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
PDDIPA+Helvetica                                  Type 1    No        WinAnsiEncoding
PDDJAB+ZapfDingbats                               Type 1    Yes       Custom
PDDJAC+Helvetica-Oblique                          Type 1    No        WinAnsiEncoding
PDDJAD+Helvetica-Bold                             Type 1    No        WinAnsiEncoding
PDDJCD+Helvetica-Condensed-Black                  Type 1    No        WinAnsiEncoding
PDDJCE+FranklinGothic-Demi                        Type 1    No        WinAnsiEncoding
PDDJDF+Symbol                                     Type 1    Yes       Custom
PDDJEF+Helvetica-Black                            Type 1    No        WinAnsiEncoding
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
Helvetica                                         Type 1    No        MacRomanEncoding
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
Helvetica                                         Type 1    No        MacRomanEncoding
unknown                                           Type CID  Yes       Adobe-Japan1
.name                                              type      unicode   encoding
------------------------------------------------------------------------------------------
Helvetica                                         Type 1    No        MacRomanEncoding
unknown                                           Type CID  Yes       Adobe-Japan1
.......
======================================================================
ERROR: test_commandlineargumenthandler_encrypted (tests.test_pdffonts.TestPdffonts)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/axel/Documents/uni/maint/repo/tests/test_pdffonts.py", line 81, in test_commandlineargumenthandler_encrypted
    commandlineargumenthandler(["tools/pdffonts.py", "-P", "foo", "samples/encryption/aes-128-m.pdf"], usage=usage),
  File "/home/axel/Documents/uni/maint/repo/tools/pdffonts.py", line 96, in commandlineargumenthandler
    pdfconversion(
  File "/home/axel/Documents/uni/maint/repo/tools/pdffonts.py", line 21, in pdfconversion
    for page in PDFPage.get_pages(
  File "/home/axel/Documents/uni/maint/repo/pdfminer/pdfpage.py", line 131, in get_pages
    doc = PDFDocument(parser, password=password, caching=caching)
  File "/home/axel/Documents/uni/maint/repo/pdfminer/pdfdocument.py", line 600, in __init__
    self._initialize_password(password)
  File "/home/axel/Documents/uni/maint/repo/pdfminer/pdfdocument.py", line 623, in _initialize_password
    raise PDFEncryptionError("Unknown algorithm: param=%r" % param)
pdfminer.pdfdocument.PDFEncryptionError: Unknown algorithm: param={'Filter': /'Standard', 'V': 4, 'CF': {'StdCF': {'Length': 16, 'AuthEvent': /'DocOpen', 'CFM': /'AESV2'}}, 'EncryptMetadata': False, 'Length': 128, 'R': 4, 'O': b'\xd0H\xd1R\x9eS]\x18\x84\xcd8V6{\x18KJ\x90\xdf\x01\xe67\xd1n\xca\x06[\xafNd\x90\x0b', 'U': b'\x95\x8cI\x13\xc0[ $\x06\x01e\xea\x14]\x88\x8d\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'P': -4, 'StrF': /'StdCF', 'StmF': /'StdCF'}
```
Perhaps there's something wrong on my end?

Reply:
Both github actions and my local machine passes this test. Are you sure that you have all requirements from requirements.txt installed?